File: loader/src/main/java/org/cubewhy/lunarcn/loader/api/ModInitializer.java
package org.cubewhy.lunarcn.loader.api;

import net.minecraft.crash.CrashReport;

public interface ModInitializer {
    default void onPreInit() {}
    default void onInit() {}
    default void onStart() {}
    default void onCrash(CrashReport crashReport) {}
    default void onStop() {}
}

File: loader/src/main/kotlin/org/cubewhy/lunarcn/loader/api/ModInitializer.kt
package org.cubewhy.lunarcn.loader.api

import net.minecraft.crash.CrashReport

interface ModInitializer {
    fun onPreInit() {}
    fun onInit() {}
    fun onStart() {}
    fun onCrash(crashReport: CrashReport) {}
    fun onStop() {}
}
